Bugsnag/Breadcrumbs/BugsnagBreadcrumbs.m

```diff
@@ -23,28 +23,38 @@ + (instancetype _Nullable)breadcrumbWithBlock:
 + (instancetype _Nullable)breadcrumbFromDict:(NSDictionary *_Nonnull)dict;
 @end
 
+#pragma mark -
+
 @interface BugsnagBreadcrumbs ()
+
 @property BugsnagConfiguration *config;
+@property NSArray<BugsnagBreadcrumb *> *breadcrumbs;
+@property NSUInteger lastFileNumber;
+@property NSUInteger maxBreadcrumbs;
+
 @end
 
 #pragma mark -
 
 @implementation BugsnagBreadcrumbs
 
 - (instancetype)initWithConfiguration:(BugsnagConfiguration *)config {
-    static NSString *const BSGBreadcrumbCacheFileName = @"bugsnag_breadcrumbs.json";
-    if (self = [super init]) {
-        _config = config;
-        _breadcrumbs = [NSMutableArray arrayWithCapacity:config.maxBreadcrumbs];
-        _readWriteQueue = dispatch_queue_create("com.bugsnag.BreadcrumbRead",
-                                                DISPATCH_QUEUE_SERIAL);
-        NSString *cacheDir = [NSSearchPathForDirectoriesInDomains(
-                                 NSCachesDirectory, NSUserDomainMask, YES) firstObject];
-        if (cacheDir != nil) {
-            _cachePath = [cacheDir stringByAppendingPathComponent:
-                             BSGBreadcrumbCacheFileName];
-        }
+    if (!(self = [super init])) {
+        return nil;
     }
+    
+    _config = config;
+    _breadcrumbs = [NSArray array];
+    // Capture maxBreadcrumbs to protect against config being changed after initialization
+    _maxBreadcrumbs = config.maxBreadcrumbs;
+    
+    NSError *error = nil;
+    NSString *cachesDir = NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES).firstObject;
```

Comment:
There's code for this in https://github.com/bugsnag/bugsnag-cocoa/blob/master/Bugsnag/Helpers/BSGCachesDirectory.m#L38